### PR TITLE
TJS-37: lesson page components (Video Overlays)

### PR DESCRIPTION
**Pull request by ltclmbdev**
This PR adds styles for Default, Blocked and Finished video overlays + custom Complete&Continue and Play From Beginning handlers. Problems I to want solve:

1. Finished and Default overlays: the "next lesson" and "play from beginning" urls structure is hardcoded [here](https://github.com/skillrecordings/products/blob/main/packages/skill-lesson/utils/handle-play-from-beginning.ts) and [here](https://github.com/skillrecordings/products/blob/main/packages/skill-lesson/video/video-overlays.tsx#L654). So I had to pass custom handlers.
2. Blocked overlay: blocked overlay should show to user the product he might want to upgrade to get the lesson available. Problem was that in TJS each playlist could belong to few products, so it was not clear what exactly product to show. It was decided to set the "Testing Pro" product as default.

<details>
  <summary>Default overlay</summary>
  
</details>
<details>
  <summary>Default overlay (mobile)</summary>
  
</details>
<details>
  <summary>Finished overlay</summary>
  
</details>
<details>
  <summary>Finished overlay (mobile)</summary>
  
</details>
<details>
  <summary>Blocked overlay</summary>
  
</details>
<details>
  <summary>Blocked overlay (mobile)</summary>
  
</details>

nbc GIF by The Voice

### CODE REVIEW

File: apps/testing-javascript/src/templates/lesson-template.tsx [hunk 1]
```diff
@@ -5,76 +5,107 @@ import MuxPlayer, {
...
-        </h2>
+      <div className="container max-w-6xl">
+        <section className="flex">
+          <div className="grow">
+            <Video
+              ref={muxPlayerRef}
+              product={defaultProduct}
+              exerciseOverlayRenderer={() => <div>DIIIIIV</div>}
+              loadingIndicator={<Spinner />}
+              customContinueHandler={customContinueHandler}
```
Comment (DrShpongle) [inline apps/testing-javascript/src/templates/lesson-template.tsx:53]: pass two custom handlers

File: apps/testing-javascript/src/trpc/routers/products.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  }),
+  getDefaultProduct: publicProcedure.query(async () => {
+    // const {getProduct} = getSdk()
+    // const products = await getProduct({
+    //   where: {
+    //     id: process.env.NEXT_PUBLIC_DEFAULT_PRODUCT_ID,
+    //   },
+    // })
+    const defaultProduct = await getActiveProduct(
+      process.env.NEXT_PUBLIC_DEFAULT_PRODUCT_ID as string,
```
Comment (DrShpongle) [inline apps/testing-javascript/src/trpc/routers/products.ts:19]: get Testing Pro product as default here (to show as upgrade purpose in Blocked Overlay)

File: packages/skill-lesson/video/video.tsx [hunk 3]
```diff
@@ -21,6 +21,8 @@ type VideoProps = {
   product?: SanityProduct
   exerciseOverlayRenderer: () => void
   loadingIndicator: React.ReactElement
+  customContinueHandler?: any
+  customPlayFromBeginningHandler?: any
```
Comment (DrShpongle) [inline packages/skill-lesson/video/video.tsx:25]: not sure how to type these handlers

File: apps/epic-web/src/components/exercise-overlay.tsx [hunk 4]
```diff
@@ -27,7 +26,8 @@ const ExerciseOverlay = () => {
 export default ExerciseOverlay
 
 const Actions = () => {
-  const {nextExercise, path, handlePlay, muxPlayerRef} = useMuxPlayer()
+  const {nextExercise, path, handlePlay, muxPlayerRef, handleContinue} =
```
Comment (joelhooks) [inline apps/epic-web/src/components/exercise-overlay.tsx:29]: moved the handlers to our video player context

File: apps/testing-javascript/src/templates/lesson-template.tsx [hunk 5]
```diff
@@ -41,6 +41,8 @@ const LessonTemplate = () => {
     <VideoProvider
       muxPlayerRef={muxPlayerRef}
       exerciseSlug={router.query.lesson as string}
+      handleContinue={customContinueHandler}
+      handlePlayFromBeginning={customPlayFromBeginningHandler}
```
Comment (joelhooks) [inline apps/testing-javascript/src/templates/lesson-template.tsx:45]: here we can provide overrides and sensible default is applied, this means we do it the same and call same handlers for all products

File: packages/skill-lesson/video/video-overlays.tsx [hunk 6]
```diff
@@ -165,23 +161,14 @@ const DefaultOverlay = ({customContinueHandler}: any) => {
...
-                          section,
-                        })
+                    handleContinue({
+                      router,
+                      module,
+                      nextExercise,
+                      handlePlay,
+                      path,
+                      section,
+                    })
```
Comment (joelhooks) [inline packages/skill-lesson/video/video-overlays.tsx]: no logic required

File: packages/skill-lesson/hooks/use-mux-player.tsx [hunk 7]
```diff
@@ -62,6 +96,8 @@ export const VideoProvider: React.FC<
   onEnded = async () => {},
   onModuleEnded = async () => {},
   onModuleStarted = async () => {},
+  handleContinue = defaultHandleContinue,
+  handlePlayFromBeginning = defaultHandlePlayFromBeginning,
```
Comment (joelhooks) [inline packages/skill-lesson/hooks/use-mux-player.tsx:100]: defaults provided here

**Review by joelhooks — APPROVED** [review]

**Pull request merged**